fundamental-concepts/choices-role-pattern/2_api_to_choices.md

```diff
@@ -0,0 +1,110 @@
+First, some good news: The DAML ecosystem will generate the `createUser` endpoint automatically and
+there is nothing to do for us. In fact, it will provide a general `create` endpoint for all defined
+contract templates in your DAML model. For the provided HTTP JSON API this is the `v1/create`
+[endpoint](https://docs.daml.com/json-api/index.html#http-request). It will take the template name
+and its arguments as POST arguments to the call. In addition, there will also be an `archive`
+endpoint at `v1/archive` to archive an active contract and mark it inactive.
+
+Try to create `Alice`'s `User` contract by running a `create` request:
+
+```
+curl -X POST -H "Content-Type: application/json" -H 'Authorization: Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJodHRwczovL2RhbWwuY29tL2xlZGdlci1hcGkiOnsibGVkZ2VySWQiOiJNeUxlZGdlciIsImFwcGxpY2F0aW9uSWQiOiJmb29iYXIiLCJhY3RBcyI6WyJBbGljZSJdfX0.VdDI96mw5hrfM5ZNxLyetSVwcD7XtLT4dIdHIOa9lcU' -d '{
+  "templateId": "User:User",
+  "payload": {
+    "username": "Alice",
+    "following": ["Bob"]
+  }}' localhost:7575/v1/create
+```{{execute T1}}
+
```

Comment:
```suggestion

Once you get the response in terminal *make sure to copy the contractID* from the response and save it in a local editor. You'll need it in the next steps. 

```

Reply:
This will go away once I figured out how to automate this.